### ISSUE TITLE
Consider licensing issues

### ISSUE THREAD

**Issue by mattico** [PistonDevelopers/hematite_server#114]
This wasn't an issue when hematite was a tech demo for the piston engine, but since it seems like there's some momentum behind making a real Minecraft implementation (which is awesome btw :+1:) this probably needs to be discussed. Also: I am not a lawyer, this is not legal advice.

If this project ever becomes successful it might be targeted by Mojang/_Microsoft_, and if they can find any evidence that someone used decompiled Minecraft code to create part of this could all be burned to the ground by lawyers. Perhaps this will never be a problem either because this project never becomes big enough to be threatening or because Mojang just doesn't care. However, the Bukkit minecraft server [has had problems](https://bukkit.org/threads/mojang-and-the-bukkit-project.309715/) with this and [TrueCraft seems to think its an issue](https://github.com/SirCmpwn/TrueCraft#get-involved). The bukkit situation was a bit different since they were basically copy-pasting decompiled java code, but the only really legal way to do this is a [clean room](https://en.wikipedia.org/wiki/Clean_room_design) implementation; no one who has read the Minecraft source can contribute code.

I might be a bit paranoid about this, but I thought I'd bring it up.

**Comment by bvssvni** [comment]
Thanks for bringing this up! The clean room implementation sounds nice.

I think such cases should be treated carefully and professionally. The only thing that matters is to **provide the evidence beyond reasonable doubt**. As long a company does this, they will be helped and get high priority to get their legal issue resolved.

If Mojang/Microsoft or any other company think there will be even a slightly possible legal case here for the future direction of the project, then I prefer them to take contact with me now (contact details is on [my Github profile](http://github.com/bvssvni/)) or start a thread on the [forum](http://discourse.piston.rs/). It is better to do this now than waiting until an unfortunate situation occurs.

**Comment by PaulBGD** [comment]
I don't think this is an issue: see glowstone, which has been around for 6+ years https://github.com/GlowstoneMC/Glowstone

**Comment by roblabla** [comment]
So long as this was done without looking at the minecraft source code, it's ok. This is what wiki.vg and other similar sources are for. So long as you use those "indirect" resources, you're fine.
